core/frontend/src/render/RenderSystem.ts

```diff
@@ -30,7 +32,7 @@ import { RenderGraphic, RenderGraphicOwner } from "./RenderGraphic";
 import { RenderMemory } from "./RenderMemory";
 import { RenderTarget } from "./RenderTarget";
 import { ScreenSpaceEffectBuilder, ScreenSpaceEffectBuilderParams } from "./ScreenSpaceEffectBuilder";
-import { ToolAdmin } from "../tools/ToolAdmin";
+import { RealityMeshGeometry } from "./webgl/RealityMesh";

```

Comment:
Nothing in the webgl directory should escape. Make an opaque type that exposes just the API you need, like you had with RenderTerrainMeshGeometry.

Reply:
Fixed.